libminifi/include/core/Processor.h

```diff
@@ -305,6 +311,19 @@ class Processor : public Connectable, public ConfigurableComponent, public std::
   Processor &operator=(const Processor &parent);
 
  private:
+  static std::mutex& getGraphMutex() {
+    static std::mutex mutex{};
+    return mutex;
+  }
+
+  // must hold the graphMutex
+  void updateReachability(const std::lock_guard<std::mutex>& graph_lock, bool force = false);
```

Comment:
Do we want to hide these functions from actual processor implementations?
I can accept that as a design decision, just interested in the motivation.

Reply:
if it turns out that there is a legit use-case where we would need to expose this method, I'm more than happy to make it non-private but I couldn't come up with a reason to do so

Reply:
Okay, I'm fine with this, thanks!